### ISSUE TITLE
920470 false positive with nextcloud exclusion rules, maybe non working 9003330

### ISSUE THREAD

**Issue by birdynm** [coreruleset/coreruleset#2434]
### Description

With a Nextcloud installation and activated Nextcloud-exclusion-rules their is a false positive alarm of rule "920470" when a calendar entry is changed in the web-interface. (I think Rule "9003330" should prevent this but didn't work, at least for me)

I tried to change rule 9003330 to phase:1 (didn't solve the Problem)

Current workaround for me is to add "ctl:ruleRemoveById=920470" to rule 9003331.

### Audit Logs / Triggered Rule Numbers
```
[16/Mar/2022:13:08:46 +0100] [x.x.x.x/sid#b3710f40][rid#b3ae9058][/cloud/remote.php/dav/calendars/user/persnlich/597EB513-B967-439E-A1AA-CBB187FB57C9.ics][2] Warning. String match "/remote.php/dav/calendars/" at REQUEST_FILENAME. [file "/usr/share/modsecurity-crs/rules/REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf"] [line "327"] [id "9003330"] [ver "OWASP_CRS/3.3.0"]
[16/Mar/2022:13:08:46 +0100] [x.x.x.x/sid#b3710f40][rid#b3ae9058][/cloud/remote.php/dav/calendars/user/persnlich/597EB513-B967-439E-A1AA-CBB187FB57C9.ics][2] Warning. String match "/remote.php/dav/calendars/" at REQUEST_FILENAME. [file "/usr/share/modsecurity-crs/rules/REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf"] [line "336"] [id "9003331"]
[16/Mar/2022:13:08:46 +0100] [x.x.x.x/sid#b3710f40][rid#b3ae9058][/cloud/remote.php/dav/calendars/user/persnlich/597EB513-B967-439E-A1AA-CBB187FB57C9.ics][2] Warning. Match of "rx ^[\\w/.+-]+(?:\\s?;\\s?(?:action|boundary|charset|type|start(?:-info)?)\\s?=\\s?['\"\\w.()+,/:=?<>@-]+)*$" against "REQUEST_HEADERS:Content-Type" required. [file "/usr/share/modsecurity-crs/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf"] [line "932"] [id "920470"] [msg "Illegal Content-Type header"] [data "text/calendar; charset=utf-8; component=vevent; charset=utf-8"] [severity "CRITICAL"] [ver "OWASP_CRS/3.3.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-protocol"] [tag "paranoia-level/1"] [tag "OWASP_CRS"] [tag "capec/1000/255/153"] [tag "PCI/12.1"]
```

### Your Environment

* CRS version (e.g., v3.3.0):
* Paranoia level setting: 1
* ModSecurity version (e.g., 2.9.3):
* Web Server and version (e.g., apache 2.4.52):
* Operating System and version: Raspbian Bullseye

### Confirmation

[x ] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by airween** [comment]
Hi @birdynm,

thanks for reporting the issue.

Looks like rule [9003330](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf) and [9003331](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-903.9003-NEXTCLOUD-EXCLUSION-RULES.conf#L322-L331) does not affected by the issue.

The FP caused by rule [920470](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf#L915-L932). As you can see, the `Content-Type` header in your case isn't allowed.

Please check your pattern: `text/calendar; charset=utf-8; component=vevent; charset=utf-8`. [Here](https://regex101.com/r/SqaGet/1) is a regex101 link, where you can check the pattern and the subjects.

For more information, you have to read the comments before the [rule](https://github.com/coreruleset/coreruleset/blob/v3.3/master/rules/REQUEST-920-PROTOCOL-ENFORCEMENT.conf#L910-L913).

I don't know this CT header is valid or not. If it is, then as you wrote, the only solution (in case of this client) is you disable the rule `920470`.

**Comment by airween** [comment]
> I have never seen this header, but searching for it gives lots of results for `text/calendar; charset=utf-8; component=vevent`. It seems a common Content-Type for CalDAV.

Thanks for researching,
 
> We could add `component` to rule 920470 in order to allow it.

Sure - but I'm afraid the only thing what user can do is add an exception.

**Comment by lifeforms** [comment]
This is now fixed by PR #2505.
